contrib/jemalloc-cmake/CMakeLists.txt

```
@@ -1,10 +1,13 @@
-if (SANITIZE OR NOT (ARCH_AMD64 OR ARCH_ARM OR ARCH_PPC64LE) OR NOT (OS_LINUX OR OS_FREEBSD OR OS_DARWIN))
+if (SANITIZE OR NOT (
+    ((OS_LINUX OR OS_FREEBSD) AND (ARCH_AMD64 OR ARCH_ARM OR ARCH_PPC64LE)) OR
+    (OS_DARWIN AND ARCH_AMD64 AND CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
```

If jemalloc is expected to work with `RelWithDebInfo` arm64 macOS builds too, then it makes sense to change to this:
```suggestion
    (OS_DARWIN AND CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
```